FBTweak/FBTweak.m

```diff
@@ -175,38 +203,59 @@ - (void)setCurrentValue:(FBTweakValue)currentValue
     }
   }
 
-  if (_currentValue != currentValue) {
-      
-    for (id<FBTweakObserver> observer in [_observers setRepresentation]) {
-      if ([observer respondsToSelector:@selector(tweakWillChange:)]) {
-        [observer tweakWillChange:self];
-      }
-    }
-      
-    _currentValue = currentValue;
-    // we can't store UIColor to the plist file. That is why we archive value to the NSData.
-    [[NSUserDefaults standardUserDefaults] setObject:[NSKeyedArchiver archivedDataWithRootObject:_currentValue] forKey:_identifier];
+  [super setCurrentValue:currentValue];
+}
 
-    for (id<FBTweakObserver> observer in [_observers setRepresentation]) {
-      [observer tweakDidChange:self];
-    }
-  }
+- (void)reset {
+  [super reset];
+  self.currentValue = nil;
 }
 
-- (void)addObserver:(id<FBTweakObserver>)observer
+@end
+
+@implementation FBPersistentTweak
+
+- (instancetype)initWithCoder:(NSCoder *)coder
 {
-  if (_observers == nil) {
-    _observers = [NSHashTable weakObjectsHashTable];
+  NSString *identifier = [coder decodeObjectForKey:@"identifier"];
+  NSString *name = [coder decodeObjectForKey:@"name"];
+  NSString *defaultValue = [coder decodeObjectForKey:@"defaultValue"];
+
+  if ((self = [self initWithIdentifier:identifier name:name defaultValue:defaultValue])) {
+    if ([coder containsValueForKey:@"possibleValues"]) {
+      self.possibleValues = [coder decodeObjectForKey:@"possibleValues"];
+    } else {
+      // Backwards compatbility for before possibleValues was introduced.
+      FBTweakValue minimumValue = [coder decodeObjectForKey:@"minimumValue"];
+      FBTweakValue maximumValue = [coder decodeObjectForKey:@"maximumValue"];
+      self.possibleValues = [[FBTweakNumericRange alloc] initWithMinimumValue:minimumValue maximumValue:maximumValue];
+    }
+
+    self.precisionValue = [coder decodeObjectForKey:@"precisionValue"];
+    self.stepValue = [coder decodeObjectForKey:@"stepValue"];
+
+    // Fall back to the user-defaults loaded value if current value isn't set.
+    self.currentValue = [coder decodeObjectForKey:@"currentValue"] ?: nil;
   }
-  
-  NSAssert(observer != nil, @"observer is required");
-  [_observers addObject:observer];
+
+  return self;
 }
 
-- (void)removeObserver:(id<FBTweakObserver>)observer
+- (void)encodeWithCoder:(NSCoder *)coder
 {
-  NSAssert(observer != nil, @"observer is required");
-  [_observers removeObject:observer];
+  [coder encodeObject:self.identifier forKey:@"identifier"];
```

Comment:
I doubt whether the key can be specified as follows: `@keypath(self, identifier)`. Cause in Foundation's repo that's how it's written.

Reply:
This is an addition by `extObjC` which we don't have here.